cubic-server/ai/wandering.cpp

```diff
@@ -0,0 +1,57 @@
+#include "wandering.hpp"
+#include "Dimension.hpp"
+#include "entities/Entity.hpp"
+#include "logging/logging.hpp"
+#include "math/Vector3.hpp"
+#include "utility/PseudoRandomGenerator.hpp"
+#include "world_storage/Level.hpp"
+
+namespace ai {
+Wandering::Wandering(Entity &entity):
+    AI(entity)
+{
+}
+
+Wandering::~Wandering() { }
+
+bool Wandering::see() { return true; }
+
+void Wandering::think()
+{
+    Vector3<double> actualPos = this->_entity.getPosition();
+
+    int x = utility::PseudoRandomGenerator::getInstance()->generateNumber(-9, 9);
+    int z = utility::PseudoRandomGenerator::getInstance()->generateNumber(-9, 9);
+
+    Vector3<double> finalPos = {actualPos.x + x, actualPos.y, actualPos.z + z};
+
+    if (!_entity.getDimension()->hasChunkLoaded(transformBlockPosToChunkPos(finalPos.x), transformBlockPosToChunkPos(finalPos.z)))
+        return;
+    if (_entity.getDimension()->getBlock({int(finalPos.x), int(finalPos.y) - 1, int(finalPos.z)}) == 0 ||
+        _entity.getDimension()->getBlock({int(finalPos.x), int(finalPos.y), int(finalPos.z)}) != 0)
```

Comment:
Somehow that if doesn't work as intended, as I can see the chicken just flying around.
Also you should check if the entity is not out of bounds because it might crash if you substract on the lowest Y coordinates.